CMakeLists.txt.in

```diff
@@ -0,0 +1,298 @@
+
+ExternalProject_Add(dpdk_ext
+  URL https://github.com/scylladb/dpdk/archive/a1774652fbbb1fe7c0ff392d5e66de60a0154df6.tar.gz
+  URL_MD5 2af337f9ed29955362845808d6cdaa3e
+  CONFIGURE_COMMAND
+    COMMAND
+      ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>
+      make ${dpdk_args} config
+    COMMAND
+      ${CMAKE_COMMAND}
+      -DSeastar_DPDK_CONFIG_FILE_IN=<BINARY_DIR>/.config
+      -DSeastar_DPDK_CONFIG_FILE_CHANGES=@CMAKE_CURRENT_SOURCE_DIR@/dpdk_config
+      -DSeastar_DPDK_CONFIG_FILE_OUT=<BINARY_DIR>/${dpdk_quadruple}/.config
+      -P @CMAKE_CURRENT_SOURCE_DIR@/cmake/dpdk_configure.cmake
+  BUILD_COMMAND ""
+  INSTALL_COMMAND
+   ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>
+   ${make_command} ${dpdk_args} install)
+
+ExternalProject_Add(c-ares_ext
+  URL https://c-ares.haxx.se/download/c-ares-1.13.0.tar.gz
+  URL_MD5 d2e010b43537794d8bedfb562ae6bba2
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> --srcdir=<SOURCE_DIR>
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ${make_command} install)
+
+ExternalProject_Add(yaml-cpp_ext
+  URL https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-0.5.3.tar.gz
+  URL_MD5 2bba14e6a7f12c7272f87d044e4a7211
+  INSTALL_DIR    @SMF_DEPS_INSTALL_DIR@
+  CMAKE_ARGS
+    -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+    -DCMAKE_PREFIX_PATH=<INSTALL_DIR>
+    -DYAML_CPP_BUILD_TESTS=OFF
+    -DBUILD_SHARED_LIBS=ON
+  BUILD_COMMAND ${make_command}
+  DEPENDS boost_ext)
+
+ExternalProject_Add(gmp_ext
+  URL https://gmplib.org/download/gmp/gmp-6.1.2.tar.bz2
+  URL_MD5 8ddbb26dc3bd4e2302984debba1406a5
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND
+    <SOURCE_DIR>/configure
+    --prefix=<INSTALL_DIR>
+    --srcdir=<SOURCE_DIR>
+    ${info_dir}/gmp
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ${make_command} install)
+
+ExternalProject_Add(nettle_ext
+  URL https://ftp.gnu.org/gnu/nettle/nettle-3.4.tar.gz
+  URL_MD5 dc0f13028264992f58e67b4e8915f53d
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND
+    <SOURCE_DIR>/configure
+    --prefix=<INSTALL_DIR>
+    --srcdir=<SOURCE_DIR>
+    --libdir=<INSTALL_DIR>/lib
+    ${info_dir}/nettle
+    ${autotools_flags}
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ${make_command} install
+  DEPENDS gmp_ext)
+
+ExternalProject_Add(gnutls_ext
+  URL https://www.gnupg.org/ftp/gcrypt/gnutls/v3.5/gnutls-3.5.18.tar.xz
+  URL_MD5 c2d93d305ecbc55939bc2a8ed4a76a3d
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND
+   ${CMAKE_COMMAND} -E env ${PKG_CONFIG_PATH}
+    <SOURCE_DIR>/configure
+    --prefix=<INSTALL_DIR>
+    --srcdir=<SOURCE_DIR>
+    --with-included-unistring
+    --with-included-libtasn1
+    --without-p11-kit
+    # https://lists.gnupg.org/pipermail/gnutls-help/2016-February/004085.html
+    --disable-non-suiteb-curves
+    --disable-doc
+    ${autotools_flags}
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ${make_command} install
+  DEPENDS gmp_ext nettle_ext)
+
+ExternalProject_Add(lz4_ext
```

Comment:
https://github.com/smfrpc/smf/blob/master/recipe/wellknown.cmake#L73-L81

we should use the ones in wellknown.cmake instead. They have slight modifications like the yamlcpp and the boost, etc. 

lz4 has a proper cmake that works tho.

Reply:
i'm not really following. it seems as though the same version is being used in this patch, in wellknown.cmake, and seastar cooking_recipes.